### ISSUE TITLE
pip install failure on f37 and ubuntu-latest

### ISSUE THREAD

**Issue by adriansev** [xrootd/xrootd#2002]
Hi! It seems that xrootd can no longer be install through pip on a fedora 37 and on github runner ubuntu-latest ..
on f37 side trying a `python3 -m pip install --user --upgrade xrootd` gives me:
```
  exec(compile(setup_py_code, filename, "exec"))
  '"'"''"'"''"'"' % ('"'"'/home/adrian/tmp/pip-install-jv42me36/xrootd_f303b9e4ef5a4f95a1d75b1ff94207ea/setup.py'"'"',), "<pip-setuptools-caller>", "exec"))' bdist_wheel -d /home/adrian/tmp/pip-wheel-3e_i88p2
  Command output: [use --verbose to show]
  Running setup.py clean for xrootd
Failed to build xrootd
ERROR: Could not build wheels for xrootd, which is required to install pyproject.toml-based projects
```  

on the github runner side, i have a workflow where i do this:
`https://github.com/adriansev/jalien_py/blob/master/.github/workflows/codeql-analysis.yml#L61`

and i get the same error as on fedora 37, see:
https://github.com/adriansev/jalien_py/actions/runs/4812709489/jobs/8568298173#step:6:101

Any idea what happened with xrootd pip install ability?
Thanks a lot!

**Comment by adriansev** [comment]
for reference, i got to make the install work by using `pip==23.0.1 setuptools==65.5.1`

**Comment by amadio** [comment]
Dear @adriansev, we are painfully aware of this (you can see the CI is red on master at the moment). The packaging of the Python bindings are in need of an uplift, which will happen soon, see https://github.com/xrootd/xrootd/issues/1844. If I can work around the issues with latest pip and setuptools without a rewrite, I will push it for next patch release next week.

**Comment by adriansev** [comment]
Thanks a lot!!

**Comment by matthewfeickert** [comment]
> I got to make the install work by using `pip==23.0.1 setuptools==65.5.1`

I think constraining `setuptools` is unecessary as this is just running into

> * Remove `setup.py install` fallback when building a wheel failed for projects without `pyproject.toml`. ([#8368](https://github.com/pypa/pip/issues/8368))
> * When the `wheel` package is not installed, pip now uses the default build backend instead of `setup.py install` and `setup.py develop` for project without `pyproject.toml`. ([#8559](https://github.com/pypa/pip/issues/8559))

as outlined in the [`pip` `v23.1` release notes](https://pip.pypa.io/en/stable/news/#v23-1) &mdash; remember `pip` does not use SemVer but a custom CalVer like system!

@henryiii can correct me if I'm wrong, but until a modern build system can be established in Issue #1844 I see no problem with just pinning `pip` to a version that hasn't deprecated support for the legacy systems used now (i.e. `pip<23.1`).

**Comment by amadio** [comment]
I'm currently rewriting the whole build system for the Python bindings, that should come out with v5.6 next month. It will solve this and many other issues, and I plan to support binary wheels too. Cheers,

**Comment by amadio** [comment]
Fixed by #2025.
